### PR TITLE
Make major version bump plans public

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

File: docs/root/major-version.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+- Change our target platforms to .NET 4.6.1 and .NET Standard 2.0
+- Revisit our retry configuration to bring it in line with gRPC
+  "native" retry configuration.
+- Remove all obsolete members from all libraries.
+- Remove all retry-related members that are not generated under by new
+  microgenerator.
+- Where individual APIs have their own copies of common resource
+  names (such as LocationName), update them to use the common
+  resource names instead.
+
```
Comment (chrisdunelm) [inline docs/root/major-version.md]: Also: Removing method signature (flattened) overloads with nullable primitive type when a parameter is optional. These option primitive parameters are now just the primitive type.

File: docs/root/major-version.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+- November 2019:
+  - Create "final for this major version" releases for all APIs
+  - Start merging breaking changes into GAX
+- December 2019:
+  - Create alpha GAX 3.0 packages, but without releasing them to NuGet
+  - Migrate API libraries to GAX 3.0
+  - Publish alpha releases of all APIs and GAX
+- January 2020:
+  - Publish beta releases of all APIs and GAX
+  - If all goes well, publish GA releases of GAX and stable APIs
```
Comment (chrisdunelm) [inline docs/root/major-version.md]: Push this into February? I'd be happier with a little more bake time for a new major version.

File: docs/root/major-version.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+We plan on making the following changes:
+
+- Upgrade the Grpc.Core dependency to the latest 2.x release.
+- Remove the System.Interactive.Async dependency, replacing it with
+  Microsoft.Bcl.AsyncInterfaces.
+- Change our target platforms to .NET 4.6.1 and .NET Standard 2.0
+- Revisit our retry configuration to bring it in line with gRPC
+  "native" retry configuration.
+- Remove all obsolete members from all libraries.
+- Remove all retry-related members that are not generated under by new
```
[the remark this answers is not in the document]
Comment (jskeet) [reply docs/root/major-version.md]: Done.

File: docs/root/major-version.md (hunk 1 above)
Comment (jskeet) [reply docs/root/major-version.md]: Done.

File: docs/root/major-version.md (hunk 2 above)
Comment (jskeet) [reply docs/root/major-version.md]: Done, although I'm less sure - that means it'll probably be about 3 months between GA releases. I suspect we may need to do releases before then for new API features etc.

**Comment by jskeet** [comment]
Right, will publish now.

**Pull request merged**